ESOAsg/scripts/fix_sphere.py

```diff
@@ -94,19 +96,67 @@ def parser(options=None):
     parser.add_argument("-v", "--version", action="version", version=__version__)
     return parser.parse_args()
 
+def get_object_coordinate(hdr):
+    """
+    """
+    if 'OBJECT' in hdr.keys():
+        try:
+            object_coordinate = SkyCoord.from_name(str(hdr['OBJECT']).strip())
+            if 'RA' in hdr.keys() and 'DEC' in hdr.keys():
+                pointing_coordinate = SkyCoord(float(hdr['RA']), float(hdr['DEC']), unit='deg')
+                msgs.work('Testing from separation from pointing position')
+                separation = object_coordinate.separation(pointing_coordinate).arcsec
+                if separation < 120.:
+                    msgs.info('Object - Pointing separation is {0:.2f} arcsec'.format(separation))
+                    object_coordinate = None
+                else:
+                    msgs.warning('Object - Pointing separation is too large: {0:.2f} arcsec'.format(separation))
+                    msgs.warning('CRVAL not updated')
+                    object_coordinate = None
+        except name_resolve.NameResolveError:
+            msgs.warning('Object {} not recognized'.format(str(hdr['OBJECT']).strip()))
+            msgs.warning('CRVAL not updated')
+            object_coordinate = None
+    else:
+        object_coordinate = None
+    return object_coordinate
+                    
+def get_platescale_deg(hdr):
+    """
+    Reads the plate scale from the header and returns the value in 
+    deg/px. If the keyword is not there, uses a default value.
+    Input:
+        hdr: the header containing the keyword PIXTOARC
+    Output:
+        platescale value in deg, ready to be used in CD[1-2]-1
+    """
+    if 'PIXTOARC' in hdr.keys():
+        platescale_mas = hdr['PIXTOARC']
+        platescale_deg = platescale_mas/1000/3600
+        msgs.info('Reading the platescale: '+\
+                  '{0:.2f} mas/px'.format(platescale_mas))
+    else:
+        if hdr['HIERARCH ESO SEQ ARM'].startswith('IFS'):
+            platescale_mas = 7.46
+        elif hdr['HIERARCH ESO SEQ ARM'].startswith('IRDIS'):
+            platescale_mas = 12.25            
+        platescale_deg = platescale_mas/1000/3600
```

Comment:
It seems that an else is missing here. 
For the moment it should raise an error stating that the instrument is missing.

Reply:
Done